docs/recipes/endpoint-testing.md

```diff
@@ -17,21 +17,20 @@ const app = require('../app');
 
 test.before(async t => {
 	t.context.server = http.createServer(app);
-	t.context.baseUrl = await listen(t.context.server);
+	t.context.prefixUrl = await listen(t.context.server);
```

Comment:
It would actually be nicer to assign a new Got instance here instead. Then you could set all the common settings here and reuse it for each test. Example: https://github.com/sindresorhus/got#custom-endpoints

Reply:
`const {email} = await got('user', {prefixUrl: t.context.prefixUrl}).json();` could then simply be `const {email} = await t.context.fetch('user');`

Reply:
I’d like to use `got` in my tests as that’s what I’m using in my application, but I’ve tied myself in knots trying to use it for endpoint testing. This is what I’m currently doing, using `supertest` and a setup function:

```js
import test from 'ava';
import supertest from 'supertest';

import app from '../../index.js';
const request = supertest(app);

const mockResponse = async query => {
  return request.get('/micropub')
    .set('Accept', 'application/json')
    .query(query);
};

test('Returns configuration', async t => {
  const response = await mockResponse('q=config');
  t.is(response.status, 200);
});
```

Should I be using `http.createServer` and `test-listen` in place of `supertest`? I can’t seem to get the piece in the right order to get the expected response.

Reply:
> It would actually be nicer to assign a new Got instance here instead.

@sindresorhus yes, but this is an endpoint testing recipe, not a Got recipe 😉 

---

@paulrobertlloyd I'm not familiar with `supertest`, but just looking at your code snippet I don't see any mocks being set up.

FWIW I tend to use `http.createServer()` and wire up my fake server myself rather than learning additional libraries.

Reply:
> @sindresorhus yes, but this is an endpoint testing recipe, not a Got recipe 😉

But you're using Got for it, and I was showing how to my the recipe more succinct.

That's like saying I should not use `t.throws`, but instead try/catch in a recipe about testing a React app, because it's not an AVA recipe.